### ISSUE TITLE
Export onnx model error

### ISSUE THREAD

**Issue by thanhpt93** [WongKinYiu/ScaledYOLOv4#149]
This below error occur when I run export.py. Could you check it? Thank you very much!

TorchScript export failure: 
Could not export Python function call 'MishCudaFunction'. Remove calls to Python functions before export. Did you forget to add @script or @script_method annotation? If this is a nn.ModuleList, add it to __constants__:
/root/miniconda3/lib/python3.7/site-packages/mish_cuda-0.0.3-py3.7-linux-x86_64.egg/mish_cuda/__init__.py(20): forward
/root/miniconda3/lib/python3.7/site-packages/torch/nn/modules/module.py(709): _slow_forward
/root/miniconda3/lib/python3.7/site-packages/torch/nn/modules/module.py(725): _call_impl
/root/tool/ScaledYOLOv4/models/common.py(31): forward
/root/miniconda3/lib/python3.7/site-packages/torch/nn/modules/module.py(709): _slow_forward
/root/miniconda3/lib/python3.7/site-packages/torch/nn/modules/module.py(725): _call_impl
/root/tool/ScaledYOLOv4/models/yolo.py(129): forward_once
/root/tool/ScaledYOLOv4/models/yolo.py(109): forward
/root/miniconda3/lib/python3.7/site-packages/torch/nn/modules/module.py(709): _slow_forward
/root/miniconda3/lib/python3.7/site-packages/torch/nn/modules/module.py(725): _call_impl
/root/miniconda3/lib/python3.7/site-packages/torch/jit/_trace.py(940): trace_module
/root/miniconda3/lib/python3.7/site-packages/torch/jit/_trace.py(742): trace
export.py(34): <module>

**Comment by sirhikesalot** [comment]
@thanhpt93 I'm getting this same error. Given that the issue is closed, did you figure out the solution? Did you just remove the function or did you need to add it to constants?

**Comment by thanhpt93** [comment]
@EconML yes, this error occur because Mish() function not being assign in model. You can try add below code to export.py:

    # Load PyTorch model
    model = attempt_load(opt.weights, map_location=torch.device('cpu')).float()  # load FP32 model

    # Update model
    for k, m in model.named_modules():
        m._non_persistent_buffers_set = set()
        if isinstance(m, models.common.Conv) and isinstance(m.act, models.common.Mish):
            m.act = Mish()
        if isinstance(m, models.common.BottleneckCSP) or isinstance(m, models.common.BottleneckCSP2) \
                or isinstance(m, models.common.SPPCSP):
            if isinstance(m.act, models.common.Mish):
                m.act = Mish()

      model.eval()
      model.model[-1].export = True

**Comment by sirhikesalot** [comment]
Still having issues. I understand that it isn't being added, but I think there is a way to use torch.JIT.script functions here. I've been struggling to find out how to get the @torch.JIT.script annotation to work. @WongKinYiu any insight would be greatly appreciated, thx. Additionally when I replace Mish-CUDA with a different activation, like [here](https://github.com/rwightman/gen-efficientnet-pytorch/blob/master/geffnet/activations/activations_jit.py#L32), I can't get the model to train on anything higher than batch size 1

**Comment by thanhpt93** [comment]
@EconML I also having issues when convert to torch scrips but convert to onnx is success. Did you converted to onnx success?

**Comment by sirhikesalot** [comment]
Unable to export to onnx either @thanhpt93 Getting below error for onnx after recommended code change you provided. I was able to print out the model layers using that though so I'm not sure why Mish isn't getting loaded in. Question: are you using the attempt_load from models/experimental.py ? Also did you train using sync batch norm? Thanks for the help

ONNX export failure: ONNX export failed: Couldn't export Python operator MishCudaFunction

Defined at:
/opt/conda/lib/python3.6/site-packages/mish_cuda-0.0.3-py3.6-linux-x86_64.egg/mish_cuda/__init__.py(20): forward
/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py(561): _slow_forward
/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py(575): __call__
/yolo/models/common.py(34): fuseforward
/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py(561): _slow_forward
/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py(575): __call__
/yolo/models/yolo.py(129): forward_once
/yolo/models/yolo.py(109): forward
/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py(561): _slow_forward
/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py(575): __call__
/opt/conda/lib/python3.6/site-packages/torch/jit/__init__.py(394): wrapper
/opt/conda/lib/python3.6/site-packages/torch/jit/__init__.py(408): forward
/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py(577): __call__
/opt/conda/lib/python3.6/site-packages/torch/jit/__init__.py(320): _get_trace_graph
/opt/conda/lib/python3.6/site-packages/torch/onnx/utils.py(319): _trace_and_get_graph_from_model
/opt/conda/lib/python3.6/site-packages/torch/onnx/utils.py(366): _model_to_graph
/opt/conda/lib/python3.6/site-packages/torch/onnx/utils.py(530): _export
/opt/conda/lib/python3.6/site-packages/torch/onnx/utils.py(92): export
/opt/conda/lib/python3.6/site-packages/torch/onnx/__init__.py(172): export
export.py(58): <module>

**Comment by bobbilichandu** [comment]
@thanhpt93 Were you able to successfully convert to torchscript? If so, can you please share the code?
Thank you!
